Play Wanderers Campaign before Hai Reveal (#22)
UnorderedSigh/endless-sky: data/wanderer/wanderers start.txt

		not attributes uninhabited
		government Hai Republic Syndicate "Free Worlds" Neutral Independent
	destination Hai-home
	to offer
		has "Wanderers: Diplomacy Delayed: offered"
		not "event: Wanderers before Hai Reveal"
		has "event: Sayari available after Hai Reveal"
	on offer
		fail "Wanderers: Diplomacy Delayed"
		fail "Wanderers: Professor Available"
		fail "Wanderers: Meet Professor"
		conversation
			`Upon landing, you receive an official communique from the Hai government. "A diplomat is available for reassignment to first contact with the avian species you met to the north. Please meet her as soon as possible at <planet>.`
				accept

mission "Wanderers: Professor Available"
	landing
	name "Wanderers: Professor Andelena"
	description "Fly to <planet> to pick up a professor and grad student who are willing to translate for you."
	source
		not planet Hai-home
		not planet Allhome
		not attributes uninhabited
		government Hai Republic Syndicate "Free Worlds" Neutral Independent
	destination Allhome
	to offer
		has "event: A Hai professor is available to translate"
		not "Wanderers: Diplomacy: accepted"
		not "Wanderers: Diplomacy: done"
	on offer
		conversation

> 			`Your ship's terminal bleats out a pinging noise, accompanied by a message from a university on All-home. It mentions Professor Andelena the Department of Skiourology at Mongongo University. The message is otherwise short and cryptic, but it sounds like an interpreter may be waiting for you on <planet>: Professor Andalena`

Reviewer: ```suggestion
			`Your ship's terminal bleats out a pinging noise, accompanied by a message from a university on Allhome. It mentions Professor Andelena the Department of Skiourology at Mongongo University. The message is otherwise short and cryptic, but it sounds like an interpreter may be waiting for you on <planet>: Professor Andalena.`
```
This is repetitive. Maybe it's better to write the message itself.

Author: I rewrote the dialog. How does it look now?

				accept

mission "Wanderers: Meet Professor"
	landing
	name "Wanderers: Professor Andelena"
	description "Pick up Professor Andelena and her student Eenap at the <planet> spaceport, so they can serve as interpreters for you."
	source Allhome
	to offer
		has "Wanderers: Professor Available: done"
		not "Wanderers: Sayari Available After Hai Reveal: offered"
	to complete
		has "Wanderers: Diplomacy (Before Hai Reveal): offered"
	on offer
		log "Minor People" "Professor Andelena" `Professor Maindell Stormchild Andelena, IV speaks with Shroud cloud city accent and is a daughter of the rich Andelena family. Her wealthy meanderings have led her to a university on Allhome where she studies Skiorology, the Hai-studying equivalent of Anthropology.`
		log "Minor People" "Eenap" `Professor Andelena's small Hai grad student, who is interested in studying the aliens to the north as her thesis research subject.`
		conversation

> 			`It turns out Mongongo University is the size of a large town, with its own spaceport. The Department of Skiourology has five entire buildings. You enter the largest and inquire about Professor Andalena, whose office is through a labyrinth of corridors. After a few minutes of navigation, you begin to conclude you've been trapped in a Mobius strip.`

Reviewer: I still get the message, `The aliens north of Hai space seem to speak the same language. Do you want to look for someone here who could serve as an interpreter?` before I get this message.

Author: This should be fixed now. There was a missing `on accept`

			`	One of them addresses you. "Are you looking for Professor Andelena?"`
			choice
				`	"Yes."`
					goto yes
				`	"Is she in some kind of trouble?"`
				`	"I've never heard that name in my life."`
					decline
			`	"Neither she nor you are in trouble. We just want you to help us clear up some matters."`
				goto question
			label yes
			`	"There has been an incident and some miscommunications. We want you to help us clear up some matters."`
			label question
			choice
				`	"Lead on."`
				`	"No, I think I'll be on my way."`
					decline
			`	The armed Hai escort you to a secure room, where a worried Professor Andelena looks at her student, Eenap, who is in handcuffs. An stern-looking Hai woman in a similar uniform to the others seems to be in charge, and glares at Eenap before addressing you.`
			`	"Your efforts to find an interpreter have stumbled upon a suspected Unfettered spy who tried to transmit details about you and your ship to the Unfettered." She glares again at Eenap, who parries with a defiant look. "Fortunately, we shut down the operation before it got too far; you need not worry about your safety in this matter.`

> 			`	"However, we are concerned about the level of interest the Unfettered have in these aliens the head of our diplomatic corps wants to send an ambassador."`

Reviewer: Should this be like this?
```suggestion
			`	"However, we are concerned about the level of interest the Unfettered have in these aliens, so the head of our diplomatic corps would like to send an ambassador."`
```
As I'm playing, it didn't seem to take very much for the Hai to turn around and say yes.

Author: I think I need to move their offer of an interpreter to another mission. It should go like this:

1. Interpreter refused on Hai-home.
2. Wait.
3. Professor wants to see you.
4. Professor's plans fall apart when student is arrested. (That's the part you commented about.)
5. Wait.
6. Message to visit a spaceport somewhere for an interpreter. (I'll pick a planet that Hai Reveal doesn't touch much.)

Author: I've added the ~~second~~ sixth step, and the Hai are quite disgruntled about *needing* to send an interpreter. Captain Last made a thorough mess of things by not displaying patience nor respecting privacy. Because of this, the low-level diplomats wanted to send somebody else, but the head of the Hai diplomatic corps (Sayari) overruled them. Sayari chose Captain Last and Irayas for the mission.

			choice
				`	"Who is the ambassador?"`
				`	"I'm done with this cloak and dagger business. Find another courier."`
					decline
			action
				log "People" "Irayas Payaana" `A Hai ambassador who made first contact with the Syndicate and various pirate groups. Sayari, the head of the Hai diplomatic corps, chose her as the Hai ambassador to the aliens to the north of Hai space.`
				log "People" "Sayari" `The head of the Hai diplomatic corps who hand-picked Irayas Payaana as the Hai representative to the aliens north of Hai space.`

> 			`	"Myself: Irayas Payaana", she says. "I made first contact with the Syndicate, and various pirate groups years ago. My specialty is maintaining positive relations with alien groups who have unknown, and possibly hostile, objectives. Most of my work recently has been with the Unfettered, but Sayari - the head of our diplomatic corps - believes you've found a more urgent need for my skills."`

Reviewer: ```suggestion
			`	"Myself, Irayas Payaana," she says. "I made first contact with the Syndicate, and various pirate groups years ago. My specialty is maintaining positive relations with alien groups who have unknown, and possibly hostile, objectives. Most of my work recently has been with the Unfettered, but Sayari - the head of our diplomatic corps - believes you've found a more urgent need for my skills."`
```

			choice
				`	"Let's go!"`
				`	"I'm done with this cloak and dagger business. Find another courier."`
					decline
			action
				event "Wanderers before Hai Reveal"
				fail "Wanderers: Diplomacy Delayed"

> 			`	"One warning Captain <last>, from someone who has seen the darkest side of the Hai. It is imperative that you keep silent about your discovery, and any hostilities you may take part in. Should your involvement become public knowledge, I fear there may be a backlash against humans, especially in Unfettered space."`

Reviewer: ```suggestion
			`	"One warning, Captain <last>, from someone who has seen the darkest side of the Hai: It is imperative that you keep silent about your discovery, and any hostilities you may take part in. Should your involvement become public knowledge, I fear there may be a backlash against humans, especially in Unfettered space."`
```

				accept

event "Wanderers before Hai Reveal"

> 

Reviewer: <img width="1440" alt="Screenshot 2023-04-02 at 2 52 19 PM" src="https://user-images.githubusercontent.com/115441627/229328492-f9b332b5-10f0-4e4c-a9fa-f8cbeecbfbb2.png">

The screenshot has a lot of things going on. 
- There's two `Wanderer: Diplomacy` missions.
- `Wanderers: Hai Diplomat` should fail when I meet Irayas and have no destination. Right now the destination is Hai-home?
- After `Wanderers: Professor Andalena`, Andalena's actions after this dialogue with Irayas isn't told to the player, so I don't know if I've picked her up yet. Conclude this and ensure the player knows only Irayas is coming, or Andalena and Irayas, or just Andalena.

`Wanderers: Hai Diplomat` and `Hai Interpreter?` should fail when you meet Irayas / Andalena, leaving `Wanderers: Diplomacy` with a new description. This makes it easier for the player because now they know they've got the diplomat and / or the interpreter and all they have to do now is go to Vara K'chrai.

Author: They fail, but the map isn't updated until you land again. I can only "change" the `Wanderers: Diplomacy` description by having another mission with the name "Wanderers: Diplomacy." You'd still see the same problem: there would be two until you land again.

I will have to ponder what to do about this.

Author: Many of the missions are invisible now. They were just there to glue things together; the player didn't really need to see them on the mission list.

				goto end
			
			label trade
			`	They respond, and <Guide> translates, "They say that they would perhaps be willing to trade with you, although to form a broader trade agreement with humans would be a serious decision and one they cannot make lightly." She pauses to listen again, and then continues, "They also say that they will discuss among themselves whether you should be given access to Wanderer technology, but again, that will not be a swiftly made decision." She pauses as the Wanderers speak again, then continues translating. "They say that in their travels, they have met few species truly capable of living in peace, and they hope that yours is one of them, because..." Again she pauses as the Wanderers speak, then says, "Because, they hate to make new friends only to lose them again."`
				goto end
			
			label communication

> 			`	They respond, and Guide> translates, "They say that their species has encountered many other species in the past, and always has found those encounters to be valuable. Although..." She pauses to listen as the Wanderers continue speaking. "Although, they say that in their travels, far more often than they encounter living species, they instead only find themselves sifting through the refuse of a species that has destroyed itself through war or carelessness. They hope that you are not such a species, because..." Again she pauses as the Wanderers speak, then says, "Because, they hate to make new friends only to lose them again."`

Reviewer: ```suggestion
			`	They respond, and <Guide> translates, "They say that their species has encountered many other species in the past, and always has found those encounters to be valuable. Although..." She pauses to listen as the Wanderers continue speaking. "Although, they say that in their travels, far more often than they encounter living species, they instead only find themselves sifting through the refuse of a species that has destroyed itself through war or carelessness. They hope that you are not such a species, because..." Again she pauses as the Wanderers speak, then says, "Because, they hate to make new friends only to lose them again."`
```
